File: Code Generation/responses/python/claude/easy/find-special-substring-of-length-k.py
class Solution:
    def hasSpecialSubstring(self, s: str, k: int) -> bool:
        i = 0
        while i < len(s):
            j = i
            while j < len(s) and s[j] == s[i]:
                j += 1
            if j - i == k:
                return True
            i = j
        return False

File: Code Generation/responses/cpp/claude/easy/find-special-substring-of-length-k.cpp
#include <string>
using namespace std;

class Solution {
public:
    bool hasSpecialSubstring(string s, int k) {
        int n = s.size();
        int i = 0;
        while (i < n) {
            char c = s[i];
            int j = i;
            while (j < n && s[j] == c) j++;
            int len = j - i;
            if (len == k) return true;
            i = j;
        }
        return false;
    }
};